src/python/pants/option/global_options.py

```diff
@@ -874,11 +874,14 @@ def register_options(cls, register):
             "tags ('-' prefix).  Useful with ::, to find subsets of targets "
             "(e.g., integration tests.)",
         )
+        ci_var = "CI" in os.environ
         register(
             "--dynamic-ui",
             type=bool,
-            default=sys.stderr.isatty(),
-            help="Display a dynamically-updating console UI as pants runs.",
+            default=((not ci_var) and sys.stderr.isatty()),
+            help="Display a dynamically-updating console UI as pants runs. This UI is disabled "
+            "if pants detects that it is running in a continuous integration environment by checking "
+            'if the CI environment variable is "true".',
```

Comment:
"is disabled by default"

Should probably clarify the help to indicate that this is how the default value is set.